### PR TITLE
IB-3284 Add /group_polls EP

### PR DESCRIPTION
**Pull request by sam-ludlow**
https://io1dev.atlassian.net/browse/IB-3284

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -6850,6 +7014,152 @@ components:
...
+                  example: '01234567-89ab-cdef-1234-56789abcdef0'
+            group:
+              description: 'The group for this group poll.'
+              type: object
+              properties:
+                uuid:
+                  description: 'Invotra UUID of the group.'
+                  type: string
+                  format: uuid
+                  example: '01234567-89ab-cdef-1234-56789abcdef0'
```
Comment (jmorahan) [inline api.swagger.yaml]: this allows moving an existing poll to a different group - have we confirmed that this is permitted?

Comment (ParryLuke) [reply api.swagger.yaml]: This should not be allowed

File: api.swagger.yaml [hunk 2]
```diff
@@ -1736,6 +1736,159 @@ paths:
...
+        - 'group polls'
+      description: 'Provision a new group poll.'
+      responses:
+        '201':
+          description: 'The group poll was created successfully.'
+          headers:
+            Location:
+              description: 'The URL at which the newly provisioned group poll may be retrieved.'
+              schema:
+                type: string
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: put in

File: api.swagger.yaml [hunk 3]
```diff
@@ -6850,6 +7014,152 @@ components:
...
       type: object
+    group-poll-schema:
+      allOf:
+        - type: object
+          properties:
+            uuid:
+              description: 'Invotra UUID of the group poll.'
+              type: string
+              format: uuid
+              example: '01234567-89ab-cdef-1234-56789abcdef0'
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: sorted it

File: api.swagger.yaml [hunk 4]
```diff
@@ -6850,6 +7014,152 @@ components:
...
+              allOf:
+                - $ref: '#/components/schemas/group-schema'
+            options:
+              description: 'The options for this group poll.'
+              type: array
+              minItems: 2
+              maxItems: 26
+              items:
+                type: string
+              example: ['Pork', 'Beef', 'Mutton']
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: add vote_count as discussed

File: api.swagger.yaml (hunk 1 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: took out

File: api.swagger.yaml [hunk 5]
```diff
@@ -6850,6 +7015,161 @@ components:
...
+              description: 'The group for this group poll.'
+              type: object
+              properties:
+                uuid:
+                  description: 'Invotra UUID of the group.'
+                  type: string
+                  format: uuid
+                  example: '01234567-89ab-cdef-1234-56789abcdef0'
+            options:
+              description: 'The options for this group poll, with vote counts.'
```
Comment (jmorahan) [inline api.swagger.yaml]:
```suggestion
              description: 'The options for this group poll.'
```

**Comment by sam-ludlow** [comment]
re-creating PR due to conflicts

**Pull request closed without merging**